org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -319,9 +324,43 @@ public void didChangeConfiguration(DidChangeConfigurationParams params) {
 			JavaLanguageServerPlugin.logException(e.getMessage(), e);
 		}
 		pm.registerWatchers();
+		configureFormatter();
 		logInfo(">>New configuration: " + settings);
 	}
 
+	private void configureFormatter() {
+		String formaterXml = preferenceManager.getPreferences().getFormatterFileName();
+		Map<String, String> options = null;
+		if (formaterXml != null) {
+			File file = new File(formaterXml);
+			if (file.isFile()) {
+				ProfileStore profileStore = new ProfileStore();
+				String profileName = preferenceManager.getPreferences().getFormatterProfileName();
```

Comment:
maybe we should provide OOTB support for a bunch of profiles (sun, google), but that can be handles in a future enhancement